pyro/poutine/guide.py

```diff
@@ -21,22 +21,22 @@ class GuideMessenger(TraceMessenger, ABC):
     Derived classes must implement the :meth:`get_posterior` method.
     """
 
-    def __init__(self, model: Callable):
+    def __init__(self, model: Callable) -> None:
         super().__init__()
         # Do not register model as submodule
         self._model = (model,)
 
     @property
-    def model(self):
+    def model(self) -> Callable:
         return self._model[0]
 
-    def __getstate__(self):
+    def __getstate__(self) -> Dict[str, object]:
         # Avoid pickling the trace.
-        state = super().__getstate__()
-        state.pop("trace")
+        state = self.__dict__.copy()
+        del state["trace"]
         return state
 
-    def __call__(self, *args, **kwargs) -> Dict[str, torch.Tensor]:  # type: ignore[override]
+    def __call__(self, *args, **kwargs) -> Dict[str, Optional[torch.Tensor]]:  # type: ignore[override]
```

Comment:
I'm surprised this is `Optional[Tensor]` rather than just `Tensor`. It might make more sense to assert or filter in the samples loop below, so callers know they are getting tensors.

Reply:
Added assert statements in the for loop (a bit longer code but probably better than filtering with if statement)